engine/src/exec-interface-graphic.cpp

```diff
@@ -713,18 +733,47 @@ void MCGraphic::GetMarkerPoints(MCExecContext& ctxt, uindex_t& r_count, MCPoint*
 
 void MCGraphic::SetMarkerPoints(MCExecContext& ctxt, uindex_t p_count, MCPoint* p_points)
 {
+    if (p_count > 65535)
+    {
+        ctxt . LegacyThrow(EE_GRAPHIC_TOOMANYPOINTS);
+        return;
+    }
+    
+    MCPoint *t_closed_points = nullptr;
+    uint2 t_newcount = 0;
+    
+    if (flags & F_MARKER_OPAQUE)
+    {
+        t_closed_points = new MCPoint[p_count];
```

Comment:
If you look at the implementation of DoCopyPoints, then the 'new MCPoint[p_count]' array will leak - DoCopyPoints overwrites the 'r_' vars (points, npoints) as they are 'out' variables (references which are unassigned on entry to the funciton, but set on exit)